examples/prom/nonlinear_elasticity_global_rom.cpp

```diff
@@ -0,0 +1,1747 @@
+        GridFunction* nodes = &x_gf;
+        int owns_nodes = 0;
+        pmesh->SwapNodes(nodes, owns_nodes);
+
+        ostringstream mesh_name, ee_name;
+        mesh_name << "deformed." << setfill('0') << setw(6) << myid;
+        ee_name << "elastic_energy." << setfill('0') << setw(6) << myid;
+
+        ofstream mesh_ofs(mesh_name.str().c_str());
+        mesh_ofs.precision(8);
+        pmesh->Print(mesh_ofs);
+        pmesh->SwapNodes(nodes, owns_nodes);
+        ofstream velo_ofs(velo_name.str().c_str());
+        velo_ofs.precision(16);
+
+        Vector v_final(vx.GetBlock(0));
+        for (int i = 0; i < v_final.Size(); ++i)
+        {
+            velo_ofs << v_final[i] << std::endl;
+        }
+
+        ofstream pos_ofs(pos_name.str().c_str());
+        pos_ofs.precision(16);
+
+        Vector x_final(vx.GetBlock(1));
+        for (int i = 0; i < x_final.Size(); ++i)
+        {
+            pos_ofs << x_final[i] << std::endl;
+        }
+
+        ofstream ee_ofs(ee_name.str().c_str());
+        ee_ofs.precision(8);
+        oper.GetElasticEnergyDensity(x_gf, w_gf);
+        w_gf.Save(ee_ofs);
+
+    }
+
+    // 15. Calculate the relative error between the ROM final solution and the true solution.
+    if (online)
+    {
+        // Initialize FOM solution
+        Vector v_fom(v_rec->Size());
+        Vector x_fom(x_rec->Size());
+
+        ifstream fom_v_file, fom_x_file;
+
+        // Open and load file
+        fom_v_file.open(velo_name.str().c_str());
+        fom_x_file.open(pos_name.str().c_str());
+
+        v_fom.Load(fom_v_file, v_rec->Size());
+        x_fom.Load(fom_x_file, x_rec->Size());
+
+        fom_v_file.close();
+        fom_x_file.close();
+
+        // Get difference vector
+        Vector diff_v(v_rec->Size());
+        Vector diff_x(x_rec->Size());
+
+        subtract(*v_rec, v_fom, diff_v);
+        subtract(*x_rec, x_fom, diff_x);
+
+        // Get norms
+        double tot_diff_norm_v = sqrt(InnerProduct(MPI_COMM_WORLD, diff_v, diff_v));
+        double tot_diff_norm_x = sqrt(InnerProduct(MPI_COMM_WORLD, diff_x, diff_x));
+
+        double tot_v_fom_norm = sqrt(InnerProduct(MPI_COMM_WORLD,
+                                     v_fom, v_fom));
+        double tot_x_fom_norm = sqrt(InnerProduct(MPI_COMM_WORLD,
+                                     x_fom, x_fom));
+
+        if (myid == 0)
+        {
+            cout << "Relative error of ROM position (x) at t_final: " << t_final <<
+                 " is " << tot_diff_norm_x / tot_x_fom_norm << endl;
+            cout << "Relative error of ROM velocity (v) at t_final: " << t_final <<
+                 " is " << tot_diff_norm_v / tot_v_fom_norm << endl;
+        }
+    }
+
+    // 16. Free the used memory.
+    delete ode_solver;
+    delete pmesh;
+
+    totalTimer.Stop();
+    if (myid == 0) cout << "Elapsed time for entire simulation " <<
+                            totalTimer.RealTime() << endl;
+
+    MPI_Finalize();
+    return 1;
```

Comment:
It may not really matter, but we usually have `return 0` for a successful run.

Reply:
Thank you for spotting all these errors Dylan!